src/screen.c

```diff
@@ -3832,12 +3833,16 @@ void announce(int lport, int ssl, char *iface) {
 			rfbLog("possible alias:    %s::%d\n",
 			    host, lport);
 		}
+
+		if (host != iface)
+			free(host);
 	}
 }
 
 static void announce_http(int lport, int ssl, char *iface, char *extra) {
 	
 	char *host = this_host();
+	char *phost = host;
```

Comment:
```this_host()``` returns an allocated string which has to be freed explicitly, but depending on certain conditions ```host``` would get either one of the function args or a static string assigned, losing the pointer to the ```this_host()``` result. By using a separate pointer for what gets printed I make sure ```host``` will be freed in any case.